### PR TITLE
[feature] Added "auth_nocache" option to OpenVPN backend

### LINKED ISSUE #177: [feature] Add support for OpenVPN auth_nocache
The option `auth_nocache` is a simple boolean, we should add it to the OpenVPN schema (both client and server).

Look for `auth-nocache` in the [openvpn manual](https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/).

### PR DESCRIPTION
**Pull request by aagman945**
Added  ```auth_nocache``` to OpenVPN

Closes #177

### CODE REVIEW

File: netjsonconfig/backends/openvpn/schema.py [hunk 1]
```diff
@@ -445,6 +445,15 @@
                     "description": "Set output verbosity for logging and debugging",
                     "propertyOrder": 52,
                 },
+                "auth_nocache": {
+                    "title": "auth nocache",
+                    "description": "Cause OpenVPN to immediately forget username/password"
+                    "inputs after they are used",
```
Comment (pandafy) [inline netjsonconfig/backends/openvpn/schema.py:451]:
```suggestion
                    "description": (
                        "Cause OpenVPN to immediately forget username/password"
                        "inputs after they are used"
                    ),
```

**Review by pandafy — CHANGES_REQUESTED** [review]
@aagman945 this PR will need some more work before merging. 

We will need to  update the `auto_client` function of OpenVpn class to add `auth_nocache` field. 

https://github.com/openwisp/netjsonconfig/blob/a90813c263dd23e56d2a2f8ef008c7135dbfee05/netjsonconfig/backends/openvpn/openvpn.py#L23-L34

Adding `auth_nocache` in the `copy_keys` list should do it. 

We will also need to update [tests](https://github.com/openwisp/netjsonconfig/blob/master/tests/openvpn/test_backend.py) after 
making this change. In the tests, set the value of `auth_nocache` to `True`. 

The [documentation](https://github.com/openwisp/netjsonconfig/blob/master/docs/source/backends/openvpn.rst#openvpn-backend-schema) needs to be updated as well to reflect addition of this field.

**Comment by aagman945** [comment]
@pandafy should I update all the tests, or  just ```test_server_mode``` & ```test_client_mode``` .

**Review by pandafy — CHANGES_REQUESTED** [review]
Good work @aagman945, everything looks good to me. I have tested this with openwisp-controller as well. 
The only problem I see is with the commit message. The commit message does not reference the issue. Read the commit style guideline here https://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly

Also, can you squash you commits and rebase you PR on the latest master?

**Comment by aagman945** [comment]
> Good work @aagman945, everything looks good to me. I have tested this with openwisp-controller as well. The only problem I see is with the commit message. The commit message does not reference the issue. Read the commit style guideline here https://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly
> 
> Also, can you squash you commits and rebase you PR on the latest master?
@pandafy Sorry ,My bad
Done with the required changes

**Review by pandafy — CHANGES_REQUESTED** [review]
@aagman945 the build is failing!

**Review by pandafy — APPROVED** [review]
LGTM! 👍🏼 

@aagman945 the first word of the commit description should be capitalised ("Closes" and not "closes"). This time, we will update the description while merging. But if you are eager to continue contributing to OpenWISP, I ask you to please take care of these small details from next time. 

Thanks for contributing! 

Deferring merge to @nemesisdesign 😄

**Comment by aagman945** [comment]
> LGTM! 👍🏼
> 
> @aagman945 the first word of the commit description should be capitalised ("Closes" and not "closes"). This time, we will update the description while merging. But if you are eager to continue contributing to OpenWISP, I ask you to please take care of these small details from next time.
> 
> Thanks for contributing!
> 
> Deferring merge to @nemesisdesign smile

I'll take  care from next time.

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**